_plugins/misc.rb

```diff
@@ -18,23 +18,53 @@ def object_items(object)
       return object
     end
 
-    # filter a list of hashes by comma-sep'd field:value pairs
-    def data_filter(data, filters)
-      if not data.is_a?(Array) or not filters.is_a?(String)
+ 
```

Comment:
Here's the tricky and weird part. 

Takes a list/array of some data and a simple filter string. For each item... define "item" as a local Ruby variable with the value of the item itself. Then also, for each key/property on the item, define that as a local variable. Then evaluate the filter string as Ruby code with all those local variables available to use. If the filter string has something that's not in the item, catch that error, define that variable as null, then try again. When the filter string is able to be executed without errors, keep/discard the item based on whether it returns truthy/falsey.

Reply:
> some way to iterate over the variables in the expression and pre-set missing ones to nil before executing it ... and I don't know if it's possible or easy in Ruby

Afaik it's not possible. There is a `local_variables` variable you can check to get a list of the current variables in scope, and you can check for `defined?(someVar)`, but I don't think there's a way to parse a string of Ruby code and extract the local variable references in it, which is what we'd also need.

> a "safe" variable accessor that returns nil if the variable doesn't exist

While this would be safer and make my Ruby code here simpler, I think it's a no-go from me from a template philosophy standpoint. I try to keep things as clear and simple for the user as possible at (almost) any cost. I wouldn't want them to have to learn and remember a special function name (separate from standard Ruby) and repeat it anywhere they use the list component.

While very weird and ugly, I do think the method I've done is the most foolproof. It kinda just lets the language handle all the difficult parts.